pkg/execution/runner/runner.go

```diff
@@ -533,11 +534,7 @@ func (s *svc) functions(ctx context.Context, tracked event.TrackedEvent) error {
 		go func() {
 			defer wg.Done()
 			for _, t := range copied.Triggers {
-				if t.EventTrigger == nil || t.Event != evt.Name {
-					// This isn't triggered by an event, so we skip this trigger entirely.
-					continue
-				}
```

Comment:
This code is unnecessary as `FunctionsByTrigger` is already ensuring that function triggers match the event name. We do not need duplicate logic in both places.